sdk/src/types/schema.ts

```diff
@@ -0,0 +1,9 @@
+export type ItemType<TSchema extends object> = TSchema[keyof TSchema];
+
+export type PrimitiveFields<TSchema extends object, TItem extends object> = {
+	[Key in keyof TItem]: Extract<TItem[Key], ItemType<TSchema>> extends never ? Key : never;
+}[keyof TItem];
+
+export type RelationalFields<TSchema extends object, TItem extends object> = {
```

Comment:
We were way over-thinking how to extract the relational types. We already know what all the available content types are from the top-level Schema, so any nested field that has one of those top-level schema item type signatures is a relational field 🤯 No need for any wrapper type or injected `__nested__` value or anything 🥳